### DISCUSSION TITLE
problem MySQL reinstall Module

### DISCUSSION THREAD

**Discussion by Danieleeffe1** [OpenMage/magento-lts#3772] [Questions & Answers]
I have a problem with reinstalling modules and plugins in OpenMage.

I have an OM 20.1 installation that has been linked to a pre-existing DB.

I loaded the various modules and plugins into the new OM installation.

when I empty the OM cache to check the installation of these modules, it often happens that OM gives me an error.

this error is always connected to the reinstallation of the modules which seems not to be able to overwrite the old DB data.

here is an example:

a:5:{i:0;s:446:"Error in file: "/var/www/teloporto/app/code/community/YouLogix/SlotBooking/sql/youlogix_slotbooking_setup/upgrade-2.0.0-2.0.1.php" - SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '22-General' for key 'eav_attribute_group.UNQ_EAV_ATTRIBUTE_GROUP_ATTRIBUTE_SET_ID_ATTRIBUTE_GROUP_NAME', query was: INSERT INTO `eav_attribute_group` (`attribute_set_id`, `attribute_group_name`, `default_id`, `sort_order`) VALUES (?, ?, ?, ?)";i:1;s:822:"#0 /var/www/teloporto/app/code/core/Mage/Core/Model/Resource/Setup.php(628): Mage::exception()

I don't know how to read these instructions, I know nothing about MySQL and the queries that can solve these problems

do you have any suggestions?

Can anyone tell me the way to solve this problem?
Thank you.

**Comment by addison74** [comment]
Updating OpenMage does not require reinstalling extensions and modules.

To find out which extension creates trouble, disable them one by one in /app/etc/modules by editing the XML files (true => false) until you find which one creates the problem. Enabling logs is also important because errors will be recorded in the files like exceptions.log, system.log.

Installing/Uninstalling an extension is mandatory to know as a procedure for an administrator. Some developers provide all the steps for removing the files and records in the DB, but they are not always complete. I have made notes about everything I have added over OpenMage, especially for what is added into the database, both in the extension own tables and in the OpenMage tables.

If we install an extension we will have a record in core_resource about the current version. If the extension has configuration options in the Backend, then we will have records in core_config_data table. If the extension creates/modifies tables, then I look in the files stored in the sql directory, there I can find out what the extension does when it is installed.

Let's look at the mentioned error. In core_resource there is the version of the extension, which is older and then the upgrade script is run. Only that the record already exists in the eav_attribute_group table. You have to backup that table (save it as CSV), remove that row and then load any of Frontend/Backend page in the browser. The upgrade script will be run and the row will be added in the table. This time should be no errors.

**Comment by Danieleeffe1** [comment]
Thank you so much

Il mar 30 gen 2024, 18:00 ADDISON ***@***.***> ha scritto:

**Comment by kiatng** [comment]
For this error:

> a:5:{i:0;s:446:"Error in file: "/var/www/teloporto/app/code/community/YouLogix/SlotBooking/sql/youlogix_slotbooking_setup/upgrade-2.0.0-2.0.1.php" - SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '22-General' for key 'eav_attribute_group.UNQ_EAV_ATTRIBUTE_GROUP_ATTRIBUTE_SET_ID_ATTRIBUTE_GROUP_NAME', query was: INSERT INTO eav_attribute_group (attribute_set_id, attribute_group_name, default_id, sort_order) VALUES (?, ?, ?, ?)";i:1;s:822:"#0 /var/www/teloporto/app/code/core/Mage/Core/Model/Resource/Setup.php(628): Mage::exception()

It just means that the upgrade script `upgrade-2.0.0-2.0.1.php` was trying to add an entry '22-General' which was already added in the table `eav_attribute_group`. It means that you do not need to add that entry again. Depending on what other things are there in the file, and if there are no other stuff there, it may be safe to skip the update for version 2.0.1. You can edit the version in table `core_resource`, look for `youlogix_slotbooking_setup`, to skip the update, change the version to one higher than 2.0.1.

**Comment by sreichel** [comment]
Change the install script to `ON DUPLICATE KEY` could fix it too.

Finally it shoud work independent of version set in `core_resource` table.

(reset `core_resource` to a lower version should do no harm when install scripts runs multiple times)
